dandi/cli/cmd_search.py

```diff
@@ -0,0 +1,212 @@
+from pathlib import Path
+
+from SPARQLWrapper import JSON, SPARQLWrapper
+import click
+import pandas as pd
+
+from .base import map_to_click_exceptions
+
+# supported fields
+# TODO: adding more
+DANDISETS_FIELDS = {
+    "approach": ["apr", "?as dandi:approach / schema:name ?apr ."],
+    "species_id": ["sid", "?as dandi:species / schema:identifier ?sid ."],
+    "species_name": ["snm", "?as dandi:species / schema:name ?snm ."],
+}
+
+ASSETS_FIELDS = {
+    "size": ["size", "?asset schema:contentSize ?size ."],
+    "format": ["format", "?asset schema:encodingFormat ?format ."],
+    "age": ["age", "?asset prov:wasAttributedTo / dandi:age / schema:value ?age ."],
+}
+
+
+def parse_validate(ctx, param, value):
+    value_parse = []
+    # parsing elements that have multiple comma-separated values
+    for el in value:
+        value_parse += el.split(",")
+    if param.name == "select_fields":
+        if ctx.params["search_type"] == "dandisets":
+            choice_list = DANDISETS_FIELDS.keys()
+        elif ctx.params["search_type"] == "assets":
+            choice_list = ASSETS_FIELDS.keys()
+        else:
+            choice_list = None
+    else:
+        choice_list = None
+    # checking if all values are in the list of possible choices
+    for el in value_parse:
+        if choice_list and el not in choice_list:
+            ctx.fail(f"{el} is not in the list: {choice_list}")
+    return value_parse
+
+
+@click.command(help="Search TODO")
+@click.option(
+    "-F",
+    "--file",
+    help="Comma-separated list of fields to display. "
+    "An empty value to trigger a list of "
+    "available fields to be printed out",
+)
+@click.option(
+    "-t",
+    "--search_type",
+    help="Type of the search.",
+    type=click.Choice(["dandisets", "assets"]),
+)
+@click.option(
+    "-s",
+    "--select_fields",
+    help="Field name for dandisets search",
+    callback=parse_validate,
+    multiple=True,
+)
+@click.option(
+    "-f",
+    "--filter_fields",
+    help="Field name for dandisets search",
```

Comment:
help is the same as for `-s` , so what is the difference?

Reply:
also we have in ls
```
  -F, --fields TEXT               Comma-separated list of fields to display.
                                  An empty value to trigger a list of
                                  available fields to be printed out
  -f, --format [auto|pyout|json|json_pp|json_lines|yaml]
                                  Choose the format/frontend for output. If
                                  'auto', 'pyout' will be used in case of
                                  multiple files, and 'yaml' for a single
                                  file.
```

I think here we should be uniform with that -- with https://github.com/dandi/dandi-archive/issues/1659 in mind, we do adhere to MVC pattern here too, and clearly separate out the model from rendering and thus have similar `-f|--format` option which might be just an output of those "search result records".   Well -- you already have `--format` below, so we are on the right track.
May be  here we should even consider RF/reusing `ls` code -- we just need a list of records to render. they might come from different search engines etc.